pkg/lib/common.js

```diff
@@ -51,28 +51,16 @@ export function invoke_functions(functions, self, args) {
     }
 }
 
-export function iterate_data(data, callback, batch) {
-    let binary = false;
-    let len = 0;
-
-    if (!batch)
-        batch = 64 * 1024;
-
-    if (data) {
-         if (data.byteLength) {
-             len = data.byteLength;
-             binary = true;
-         } else if (data.length) {
-             len = data.length;
-         }
-    }
-
-    for (let i = 0; i < len; i += batch) {
-        const n = Math.min(len - i, batch);
-        if (binary)
-            callback(new window.Uint8Array(data.buffer, i, n));
-        else
-            callback(data.substr(i, n));
+export function iterate_data(data, callback, batch = 64 * 1024) {
+    if (typeof data === 'string') {
+        for (let i = 0; i < data.length; i += batch) {
+            callback(data.substring(i, i + batch));
```

Comment:
Thinking aloud: The old function did the `n = min(len-i, batch)` calculation for strings as well. So the idea is that i + batch can only overflow at the end, and substring() handles that correctly. So :+1:

Reply:
Yes.  :)